plutus-core/plutus-core/src/PlutusCore/Builtin/HasConstant.hs

```diff
@@ -29,6 +32,16 @@ class HasConstant term where
 -- and connects @term@ and its @uni@.
 type HasConstantIn uni term = (UniOf term ~ uni, HasConstant term)
 
+-- | Wrap a Haskell value (given its explicit type tag) as a @term@.
+fromValueOf :: HasConstant term => UniOf term (Esc a) -> a -> term
+fromValueOf uni = fromConstant . someValueOf uni
+{-# INLINE fromValueOf #-}
+
+-- | Wrap a Haskell value (provided its type is in the universe) as a @term@.
+fromValue :: (HasConstant term, UniOf term `Includes` a) => a -> term
```

Comment:
The other direction is `asConstant`. We can't have `term -> a`, because `a` would be escaped (i.e. wrapping something into an existential is fine, unwrapping from the existential is not).

So we implement the other direction as `readKnownConstant` downstream.